phoenix-spark-base/src/main/java/org/apache/phoenix/spark/datasource/v2/reader/PhoenixInputPartitionReader.java

```diff
@@ -94,6 +99,10 @@ private QueryPlan getQueryPlan() throws SQLException {
         }
         try (Connection conn = DriverManager.getConnection(
                 JDBC_PROTOCOL + JDBC_PROTOCOL_SEPARATOR + zkUrl, overridingProps)) {
+            PTable pTable = PTable.parseFrom(options.getTableBytes());
+            org.apache.phoenix.schema.PTable table = PTableImpl.createFromProto(pTable);
+            PhoenixConnection phoenixConnection = conn.unwrap(PhoenixConnection.class);
+            phoenixConnection.addTable(table, System.currentTimeMillis());
```

Comment:
Please add a log message that we are adding a cached version of the PTable here (rather than resolving the real table).

On the off chance that there is some problem (from when we read the table until when we ran this code), it would be good to remind anyone looking at the log that we're doing this trick.

Reply:
Also, do we want to record the time in which the PTable was cached and use that rather than `System.currentTimeMillis()`?

My thinking is probably "no". If we use the original time we read the PTable and a long time elapsed (really big job, starvation on the Executor side), it could be hour(s) until this code is actually executed. If the `UPDATE_CACHE_FREQUENCY` isn't increased to a large value, it might cause the Phoenix client to re-fetch the PTable again? (just guessing).

Have you thought about this case already?

Reply:
Interesting point about the timestamp.

The point of this patch is to avoid hammering the system tables with a huge number of parallel requests.
I think that if we have executor starvation, then the jobs will not be started immediately, and the syscat load is not really a problem.

Can you think of a case of when the jobs are delayed enough for this to matter, but enough of them start up synchrounously for that to be a problem (I don't know enough about Spark to tell) ?

Reply:
> Can you think of a case of when the jobs are delayed enough for this to matter, but enough of them start up synchrounously for the generated load to be a problem (I don't know enough about Spark to tell) ?

This used to be somethign that would happen in MapReduce with lots of mappers (where the state of the world might change from job submissions until the last mappers actually got scheduled and ran). Honestly, I don't think it's something we can really optimize anyways (and it's not the "common" path that people would be running `alter table` commands multiple times a day).

> the jobs will not be started immediately, and the syscat load is not really a problem.

That's a good point too. I didn't think about that side.